internal/serverinstall/serverinstall.go

```diff
@@ -0,0 +1,22 @@
+package serverinstall
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clicontext"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+	pb "github.com/hashicorp/waypoint/internal/server/gen"
+)
+
+type Installer interface {
+	Install(ctx context.Context, ui terminal.UI, log hclog.Logger) (*clicontext.Config, *pb.ServerConfig_AdvertiseAddr, string, error)
+	InstallFlags(*flag.Set)
+}
+
+var Platforms map[string]Installer = map[string]Installer{
```

Comment:
You should be able to just do this... but test it...

```suggestion
var Platforms = map[string]Installer{
```